chapters/CRUD-Operations.md

```diff
@@ -207,612 +207,539 @@ Encode the **did:btcr2** identifier as follows:
+signal. That map is then used to retrieve and validate a ::BTCR2 Update:: using
+the specific identifier being resolved as the key.
 
-It returns the following output:
+The ::Signal Bytes:: must be retrieved from the last transaction output of the
+Beacon Signal. These bytes must then be used to retrieve and validate the Beacon
+Announcement Map. A ::BTCR2 Update Announcement:: must be retrieved from the map
+document using the hex encoded SHA256 hash of the identifier being resolved as
+the key. There may be no announcement for the identifier. The announcement is a
+32-byte hash of the ::BTCR2 Update::, this must be used to retrieve and validate
+the ::BTCR2 Update::.
 
-* `btcr2Updates` - an array of ::BTCR2 Updates::
+Retrieval of the Beacon Announcement Map and ::BTCR2 Update:: documents may be
+done through querying either ::Sidecar Data:: or a ::CAS::. If querying a
+::CAS::, the 32-byte hashes must be transformed into a CID following the IPFS
+CID v1 specification.
 
-The steps are as follows:
+##### Hide {.unnumbered .unlisted}
 
-1. Set `updates` to an empty array.
-1. For `beaconSignal` in `beaconSignals`:
-    1. Set `type` to `beaconSignal.beaconType`.
-    1. Set `signalTx` to `beaconSignal.tx`.
-    1. Set `signalId` to `signalTx.id`.
-    1. Set `signalSidecarData` to `signalsMetadata[signalId]`. TODO: formalize structure of sidecarData
-   1. Set `btcr2Update` to null.
-   1. If `type` == `SingletonBeacon`:
-      1. Set `btcr2Update` to the result of passing `signalTx` and
-        `signalSidecarData` to the [Process Singleton Beacon Signal] algorithm.
-   1. If `type` == `CIDAggregateBeacon`:
-      1. Set `btcr2Update` to the result of passing `signalTx` and
-        `signalSidecarData` to the [Process CIDAggregate Beacon Signal] algorithm.
-   1. If `type` == `SMTAggregateBeacon`:
-      1. Set `btcr2Update` to the result of passing `signalTx` and
-        `signalSidecarData` to the [Process SMTAggregate Beacon Signal] algorithm.
-    1. If `btcr2Update` is not null, push `btcr2Update` to `updates`.
-1. Return `updates`.
-
-##### Confirm Duplicate Update
-
-The Confirm Duplicate Update algorithm verifies that a ::BTCR2 Update:: is a
-duplicate against the hash history of previously applied updates.
+##### Imperative Algorithm {.unnumbered .unlisted}
 
-It takes the following inputs:
+Given the 32 `signalBytes` from the Map Beacon Signal and a `sidecarDocumentsMap`:
 
-* `btcr2Update` - the ::Unsecured BTCR2 Update:: to confirm as a duplicate or not; REQUIRED; object.
-* `btcr2UpdateHashHistory` - an array of hashes corresponding to each ::BTCR2 Update::; REQUIRED; array.
+1. Set `id` to the hexadecimal string representation of `signalBytes`.
+2. Get `map` from `sidecarDocumentsMap` by its `id` if available, or from
+3. ::CAS:: by its `id` if not and `cas` is defined.
+3. If `map` is undefined, raise InvalidDidUpdate error.
+4. Set `updateId` to the value of `map.<did>`.
```

Comment:
```suggestion
4. Set `updateId` to the value of `map.<did>`.
```

We need to add a step here that first hashes the did to generate the DID index in the map.
Probably easier to do in a IDE